internal/vds/vds.cpp

```diff
@@ -312,14 +230,12 @@ void set_voxels(
     };
 
     int voxelline;
-    auto vdim   = dim_tovoxel(dimension);
-    auto system = axis_tosystem(ax);
+
+    const Axis axis(ax, layout);
+    auto vdim   = axis.get_voxel_index();
+    auto system = axis.get_coordinate_system();
     switch (system) {
         case ANNOTATION: {
-            auto transformer = OpenVDS::IJKCoordinateTransformer(layout);
```

Comment:
You are right. This should not have been removed. At least not in this larger commit. 

I am not sure how much it protects at the moment and whether it might be ambiguous.
1. For slice and slice metadata requests we check the axis order. This explicitly for looks for axes called "Inline", "Crossline" and "Sample"/"Depth"/"Time". This seems to implicitly check whether the annotation system is defined. 

   OpenVDS checks whether the start and end coordinates for the annotation system are different from each other if one calls `transformer.AnnotationsDefined`. I assume that OpenVDS sets the start and end coordinates to the zero vector as default value such that the check makes sense.
3. For slice requests we additionally check whether the annotation coordinate system is present by calling `transformer.AnnotationsDefined()`, after we have verified the order of the axes.
4. For slice request the existence of the annotation system is never checked?!
5. We check for whether the annotation system is defined in the VDS only if the **request** is using the annotation system. If the request is coming in another coordinate system, we don't bother checking whether the annotation system is present in the VDS. However, we always assume that the annotation coordinate system is present in the VDS and map the request accordingly to the annotation system's axes.

Do I miss something here? If not, maybe we should verify the existence of the annotation coordinate system when opening the VDS independent of the request.